services/worker-manager/src/providers/azure.js

```diff
@@ -387,19 +383,229 @@ class AzureProvider extends Provider {
 
   async scanPrepare() {
     this.seen = {};
+    this.errors = {};
+  }
+
+  /**
+   * Checks the status of ongoing Azure operations
+   * Returns true if the operation is in progress, false otherwise
+   *
+   * op: a URL for tracking the ongoing status of an Azure operation
+   * errors: a list that will have any errors found for that operation appended to it
+   */
+  async handleOperation({op, errors}) {
+    let req, resp;
+    try {
+      // TODO: we could be smarter about getting operation status
+      // see here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations
+      req = new msRestJS.WebResource(op, 'GET');
+      // sendLongRunningRequest polls until finished but this is just reading
+      // the status of an operation so it shouldn't block long
+      // it's ok if we hit an error here, that will trigger resource teardown
+      resp = await this._enqueue('opRead', () => this.restClient.sendLongRunningRequest(req));
+    } catch (err) {
+      // Rest API has different error semantics than the SDK
+      if (_.has(resp, 'status') && resp.status === 404) {
```

Comment:
I suspect this can never occur -- `resp` is only set *after* the `_enqueue`d operation has completed successfully, meaning that no error would be thrown.  Should this analysis of `resp` happen *outside* of the catch?

Reply:
Yes! Fixed.